projects/plugins/jetpack/modules/masterbar/admin-menu/class-base-admin-menu.php

```diff
@@ -221,6 +266,71 @@ public function enqueue_scripts() {
 		);
 	}
 
+	/**
+	 * Hide the submenu page based on slug and return the item that was hidden.
+	 *
+	 * Instead of actually removing the submenu item, a safer approach is to hide it and filter it in the API response.
+	 * In this manner we'll avoid breaking third-party plugins depending on items that no longer exist.
+	 *
+	 * A false|array value is returned to be consistent with remove_submenu_page() function
+	 *
+	 * @param string $menu_slug The parent menu slug.
+	 * @param string $submenu_slug The submenu slug that should be hidden.
+	 * @return false|array
+	 */
+	public function hide_submenu_page( $menu_slug, $submenu_slug ) {
+		global $submenu;
+
+		if ( ! isset( $submenu[ $menu_slug ] ) ) {
+			return false;
+		}
+
+		foreach ( $submenu[ $menu_slug ] as $i => $item ) {
+			if ( $submenu_slug !== $item[2] ) {
+				continue;
+			}
+
+			$this->hide_submenu_element( $i, $menu_slug, $item );
+
+			return $item;
+		}
+
+		return false;
+	}
+
+	/**
+	 * Apply the hide-if-js CSS class to a submenu item.
+	 *
+	 * @param int    $index The position of a submenu item in the submenu array.
+	 * @param string $parent_slug The parent slug.
+	 * @param array  $item The submenu item.
+	 */
+	public function hide_submenu_element( $index, $parent_slug, $item ) {
```

Comment:
Non-blocking: The thing that I struggle the most is to remember the difference between `hide_submenu_element` and `hide_submenu_page`.

What do you think about changing `hide_submenu_page` to `add_submenu_class` where the hiding class is given as an additional param? That way we can use this function for adding other classes too, but also will slightly improve the time needed to understand the logic performed here.

I'd also change the params order and remove the item param to make the function even more natural:

```php
function add_submenu_class( $slug, $index, $class ) {
  global $submenu;
  
  if ( ! isset( $submenu[ $slug ] ) || ! isset( $submenu[ $slug ][ $index ] ) ) {
    return false;
  }

  $item = $submenu[ $slug ][ $index ];

  if ( empty( $item[4] )  ) {
    $item[4] = $class;
  } else {
    $item[4] .= $class;
  }

  // phpcs:ignore WordPress.WP.GlobalVariablesOverride.Prohibited
  $submenu [ $slug ][ $index ] = $item;
}
```

Reply:
Imho, we shouldn't try to make it too generic since we don't have a need for it (yet) and I think having an abstraction over how the item is hidden it makes things easier. 

In this case, for example, we'll have to remember that in order to hide an element we'll have to apply a specific CSS, instead of simply calling the method for a specific submenu. However if we do need something more generic, we could refactor this then. :)

The purpose of the `hide_submenu_element` was to reduce duplicate code and is used as a helper for Base_Admin_Menu internal behavior.

Don't get me wrong, I do see your point, I think we should make a better distinction between these two methods. How about this:
- rename the method to something like `hide_submenu_item` would make more sense?
- mark the method with @ internal phpdoc mentioning that it's used for Base_Admin_Menu
- make the method private and moving it at the bottom the class
- remove $item argument since it's redundant.

What do you think? :)

Reply:
> having an abstraction over how the item is hidden it makes things easier.

> The purpose of the hide_submenu_element was to reduce duplicate code

While this certainly makes the coder easier to write, I'm not sure it does for reading it (which is something we spend more time on while maintaining code).

> rename the method to something like hide_submenu_item would make more sense?

I think the problem is specifically with the `hide_submenu` prefix (common to both functions) and the `page`/`item`/`element` suffix which could be quite similar in this context. E.g. what's the different between `page` and `element`/`item`? It seems so tied to the implementation, that it's hard to spot it even if we add additional changes to the docstring as suggested. 

Anyway, I know naming things is hard 🙂 , and most of the times it's just a matter of a personal preference. So unless anyone less weight in, let's leave things as they are now. This does improves the user experience fixing an important bug which is our priority at the moment.